<?xml version="1.0"?>
<launch>
  <arg name="namespace" default="center"/>
  <node
    pkg="saliency_detector"
    type="saliency_detector_node"
    name="saliency_detector"
    machine="search"
    ns="/processes/sample_detection/search/$(arg namespace)">
    <param name="img_topic" type="string" value="/cameras/search/$(arg namespace)/image_color"/>
    <remap from="/cameras/search/info" to="/cameras/search/$(arg namespace)/camera_info"/>
    <param name="bms_opening_width" type="int" value="3"/>
    <param name="bms_dilation_width_1" type="int" value="0"/>
    <param name="bms_thresh" type="double" value="20"/>
    <param name="bms_thresh_on" type="bool" value="True"/>
    <param name="bms_img_width" type="double" value="400"/>
    <param name="bms_top_trim" type="double" value="0"/>
    <param name="filterByConvexity" type="bool" value="True"/>
    <param name="filterByArea" type="bool" value="True"/>
    <param name="minArea" type="double" value="2500.0"/>
    <param name="maxArea" type="double" value="100000.0"/>
  </node>

  <node
    pkg="saliency_detector"
    type="ground_projector_node"
    name="ground_projector"
    machine="search"
    ns="/processes/sample_detection/search/$(arg namespace)">
    <param name="sub_plane_model_topic" type="string"
      value="/cameras/navigation/$(arg namespace)/planefit/model"/>
  </node>

  <node
    pkg="saliency_detector"
    type="color_histogram_descriptor_node"
    name="color_histogram_descriptor"
    machine="search"
    ns="/processes/sample_detection/search/$(arg namespace)">
    <remap from="/processes/sample_detection/search/$(arg namespace)/named_point"
      to="/processes/sample_detection/search/named_point"/>
  </node>

  <node
    pkg="topic_tools"
    type="relay"
    name="search_camera_info_relay"
    machine="search"
    ns="/processes/sample_detection/search/$(arg namespace)"
    args="/cameras/search/$(arg namespace)/camera_info /processes/sample_detection/search/kalman_filter/camera_info">
  </node>

</launch>
